### DISCUSSION TITLE
IDL_StructTagInfoByIndex

### DISCUSSION THREAD

**Discussion by motorbytes** [gnudatalanguage/gdl#1366] [Q&A]
Hi,

Is there an equivalent to the `IDL_StructTagInfoByIndex` function in GDL? I'm trying to port some code that uses a compiled C code .so shared library. However, the shared library references the above function and GDL fails when trying to call that IDL specific function because it doesn't know what it is.

Is there some kind of workaround for this - perhaps someone has already written an equivalent or something? :)

Thanks in advance for any help,

Andy Barnes.

**Comment by GillesDuvert** [comment]
Hi, interesting question. This is pushing GDL a bit too far outside its purpose and limits.. Even if GDL knows how to incorporate compiled code (by CALL_EXTERNAL) in your particular case as the C code really needs the correct output of `IDL_StructTagInfoByIndex`, then there must be a C `IDL_StructTagInfoByIndex` function in GDL that, according to the IDL documentation, returns the physical offset at which some tag lies a GDL structure. Not very difficult per se, but will not be very high in an already extremely long priority list...
You may want to dig into the C code your refer to to see if the use of  `IDL_StructTagInfoByIndex` is really necessary and cannot be replaced by something simpler --- already the simple idea of a GDL structure shared between a .so and GDL gives me goosebumps.
There is also a remote possibilty that you can write a dummy `IDL_StructTagInfoByIndex` C function that returns the correct answer (?)
HIH
